doc/src/site/sphinx/ml/sw_stacked_ensemble.rst

```diff
@@ -0,0 +1,198 @@
+            ...
+            ensemble.deleteBaseModels()
+
+        You can also get raw model details by calling the *getModelDetails()* method available on the model as:
+
+        .. code:: scala
+
+            ensembleModel.getModelDetails()
+
+        Run Predictions
+
+        .. code:: scala
+
+            ensembleModel.transform(testingDF).show(false)
+
+
+    .. tab-container:: Python
+        :title: Python
+
+        First, let's start PySparkling Shell as
+
+        .. code:: shell
+
+            ./bin/pysparkling
+
+        Start H2O cluster inside the Spark environment
+
+        .. code:: python
+
+            from pysparkling import *
+            hc = H2OContext.getOrCreate()
+
+        Parse the data using H2O and convert them to Spark Frame
+
+        .. code:: python
+
+            import h2o
+            frame = h2o.import_file("https://raw.githubusercontent.com/h2oai/sparkling-water/master/examples/smalldata/prostate/prostate.csv")
+            sparkDF = hc.asSparkFrame(frame)
+            dataset = sparkDF.withColumn("CAPSULE", sparkDF.CAPSULE.cast("string"))
+
+        Setup the algorithms the StackedEnsemble will operate with. StackedEnsemble will automatically train the corresponding
+        (base) models and pass them to H2O backend when needed. There are currently two options how a meta-learner
+        in StackedEnsemble combines the base models. It either utilizes cross validated predictions or uses a blending frame.
+        In the former case, it's important to keep the same folding across the base models and set
+        *setKeepCrossValidationPredictions* to *true* as the cross-validated predicted values will be used by meta-learner.
+        Furthermore, as the Stacked Ensemble combines the base models inside an H2O backend the base models have to be
+        available there as well and therefore *setKeepBinaryModels* has to be set to *true* too.
+
+        .. code:: python
+
+            from pysparkling.ml import H2ODRF, H2OGBM, H2OStackedEnsemble
+            drf = H2ODRF()
+            drf.setLabelCol("CAPSULE")
+            drf.setNfolds(5)
+            drf.setFoldAssignment("Modulo")
+            drf.setKeepBinaryModels(True)
+            drf.setKeepCrossValidationPredictions(True)
+
+            gbm = H2OGBM()
+            gbm.setLabelCol("CAPSULE")
+            gbm.setNfolds(5)
+            gbm.setFoldAssignment("Modulo")
+            gbm.setKeepBinaryModels(True)
+            gbm.setKeepCrossValidationPredictions(True)
+
+        Then, specify the algorithms when setting up the StackedEnsemble and train it.
+
+        .. code:: python
+
+            ensemble = H2OStackedEnsemble()
+            ensemble.setBaseAlgorithms([drf, gbm])
+            ensemble.setLabelCol("CAPSULE")
+
+            ensemble_model = ensemble.fit(dataset)
+
+        Inside the *fit* method, StackedEnsemble will internally create, utilize and delete models corresponding
```

Comment:
great tutorial, explains everything clearly, I really like it, also the API seems nice 👍 
the only thing I would try to avoid here is duplicating the text for scala/python, maybe that can be restructured somehow?

Reply:
Yeah, finally I used reST substitution mechanism to avoid text duplication.